spec/requests/api/providers_spec.rb

```diff
@@ -1116,4 +1116,14 @@ def gen_import_request
       end
     end
   end
+
+  context "#options" do
+    it "returns options for queried provider class" do
+      api_basic_authorize
+      ExtManagementSystem.descendants.select { |ems| ems.respond_to?(:provider_settings) }.each do |ems_type|
```

Comment:
@imtayadeway doing so will force me to use specific providers names as this is only implemented for "ManageIQ::Providers::Kubernetes::ContainerManager" currently. I am afraid of breaking tests when things changes in another repo (ie someone will change the way options are saved there and tests here will be red).  I also agree that the test logic is not super helpful but it helped me.
I will try to add a test with this provider and options that I don't think will change anytime soon.